File: steps/canonical/py/permissions.py
import re

# A tiny permission gate: dangerous shell commands are blocked, everything else
# runs. The real Claude Code has modes (ask / accept-edits / yolo) and layered
# rules; here we keep just the essential idea — check before you run.
#region permissions
_DANGEROUS = [
    r"\brm\s+-rf\b",
    r"\bgit\s+push\b",
    r"\bgit\s+reset\s+--hard\b",
    r"\bsudo\b",
    r"\bmkfs\b",
    r">\s*/dev/",
]


def check_permission(name: str, inp: dict) -> str:
    if name == "run_shell" and any(re.search(p, str(inp.get("command", ""))) for p in _DANGEROUS):
        return "deny"
    return "allow"
#endregion

File: steps/canonical/ts/permissions.ts
// A tiny permission gate: dangerous shell commands are blocked, everything else
// runs. The real Claude Code has modes (ask / accept-edits / yolo) and layered
// rules; here we keep just the essential idea — check before you run.
//#region permissions
const DANGEROUS = [
  /\brm\s+-rf\b/,
  /\bgit\s+push\b/,
  /\bgit\s+reset\s+--hard\b/,
  /\bsudo\b/,
  /\bmkfs\b/,
  />\s*\/dev\//,
];

export function checkPermission(name: string, input: Record<string, any>): "allow" | "deny" {
  if (name === "run_shell" && DANGEROUS.some((re) => re.test(String(input.command || "")))) {
    return "deny";
  }
  return "allow";
}
//#endregion
